Print messages about ports that can't change at runtime for K8s

Sgitario opened the pull request:
Related to https://github.com/quarkusio/quarkus/issues/33307, task 3. 
Fix https://github.com/quarkusio/quarkus/issues/32882

I didn't add this logic to Knative (because knative is renaming the port to http1) and Kind/Minikube because these are for DEV purposes only.

Sgitario commented:
@iocanel @cescoffier I'm not sure if this is what you had in mind for the task 3 of https://github.com/quarkusio/quarkus/issues/33307.
Note that I don't think this message should use the INFO level, but DEBUG.

iocanel commented on extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java:
```diff
@@ -186,20 +192,40 @@ public static Map<String, Port> combinePorts(List<KubernetesPortBuildItem> ports
+        for (Port port : allPorts) {
+            for (Map.Entry<String, PortProperty> portWithConfig : PORTS_WITH_CONFIG.entrySet()) {
+                if (port.getName().equals(portWithConfig.getKey())) {
+                    LOG.info(String.format("The target '%s' is mapping the container to the port '%s' which value is '%d'. "
+                            + "You won't be able to change it at runtime using the property '%s'.",
+                            target, port.getName(), port.getContainerPort(), portWithConfig.getValue().propertyName));
```
```suggestion
                    LOG.warn(String.format("'%s' manifests are generated with container port '%s' having value '%d'. "
                            + "The above is influenced by property '%s' which may change at runtime. Such change is likely to cause problems as the application and the container configuration will get out of sync.",
                            target, port.getName(), port.getContainerPort(), portWithConfig.getValue().propertyName));
```

Sgitario replied:
I think the message is now too long. Also, using a warning might cause some panic among users that as far as I know, never complained about this.

cescoffier replied:
let's use info instead.

iocanel reviewed (changes requested):
I found the message a bit misleading, so I suggested something different.

Also, since this `pattern` is something used in others places too, I am wondering if we should have a utility that does the config reading and logging. It would provide a more concise user experience and possibly make our lives easier years from now when we'll have forgotten all about it.

cescoffier reviewed (changes requested):
I like @iocanel suggestion.

cescoffier commented:
I agree about having a utility class with a consistent, wellformed, not misleading message.

Sgitario commented:
I'm -1 with the Ioannis' suggestion because the message is too long (imagine having the management port plus the HTTP port enabled), and I think that using the warning log level will cause some panic to users that as far as I know, never complained about this.

Sgitario commented:
After speaking with @iocanel , we agreed with changing the message to:
`[INFO] [io.quarkus.kubernetes.deployment.KubernetesDeployer] The 'kubernetes' manifests are generated with the container port 'http' having value '8080'. The app and manifests will get out of sync if the property 'quarkus.http.port' is changed at runtime.`.

PR updated

Sgitario committed:
Print messages about ports that can't change at runtime for K8s

Related to https://github.com/quarkusio/quarkus/issues/33307, task 3.
Fix https://github.com/quarkusio/quarkus/issues/32882

iocanel committed:
refactor: port mgmt from kube to ext publishing the
build items

Sgitario commented on extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java:
```diff
@@ -291,8 +291,7 @@ public List<DecoratorBuildItem> createDecorators(ApplicationInfoBuildItem applic
 
-        printMessageAboutPortsThatCantChange(KUBERNETES, ports, config);
-
+        printMessageAboutPortsThatCantChange(name, ports, config);
```
I think, name should be KUBERNETES

Sgitario reviewed (commented):
I'm -1 to include these changes as it overcomplicates the logic for very little benefit (printing messages for users that as far as I know, they never complained about it). 

Though, aside of my comments, I don't want to block this change as long as @iocanel and @cescoffier are ok with proceeding.

Sgitario commented:
@cescoffier can you take another look to this pull request after the changes made by Ioannis please?

cescoffier commented:
The CI issues might be related.

iocanel reviewed (approved):
LGTM

iocanel commented:
Don't think the failures are related.

iocanel commented:
> Don't think the failures are related.
Verified that the same tests fail in other pull requests too. So, I'll merge.
